milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
Out of curiosity, what's the difference between returning `[]` here and setting `relationship = None`? It looks like the following line already expects that on occasion.